### PR TITLE
[SPARK-47611][SQL] Cleanup dead code in MySQLDialect.getCatalystType

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR removes an unnecessary case-match branch for Types.BIT in MySQLDialect.getCatalystType, this is a special case for Maria Connector/J and can be handled in defaults as we have matched and handled Types.BIT&size > 1 before this.

Additionally, we add some new tests for this corner case and other MySQL/Maria quirks

### Why are the changes needed?

code refactoring and test improvement

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

new tests
### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

File: connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MySQLIntegrationSuite.scala [hunk 1]
```diff
@@ -150,13 +151,25 @@ class MySQLIntegrationSuite extends DockerJDBCIntegrationSuite {
     assert(rows.get(4).isInstanceOf[BigDecimal])
     assert(rows.get(5).isInstanceOf[BigDecimal])
     assert(rows.get(6).isInstanceOf[Double])
+    // Unlike MySQL, MariaDB seems not to distinguish signed and unsigned tinyint(1).
+    val isMaria = jdbcUrl.indexOf("disableMariaDbDriver") == -1
```
Comment (yaooqinn) [inline connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MySQLIntegrationSuite.scala:155]: This illustrates a deviation between MariaDB and MySQL Connector/J. It‘s not caused by removing codes in this PR.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @yaooqinn .

**Review by HyukjinKwon — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you all.

**Comment by yaooqinn** [comment]
Thank you @dongjoon-hyun @HyukjinKwon

**Pull request closed without merging**